addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/handler/DSMRMeterHandler.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.handler;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObject;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemValue;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemValueDescriptor;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeter;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterConfiguration;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterConstants;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterDescriptor;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterListener;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The MeterHandler will create logic DSMR meter ThingTypes
+ */
+public class DSMRMeterHandler extends BaseThingHandler implements DSMRMeterListener {
+    // logger
```

Comment:
Please add only useful comments and if you document fields or methods use proper docblocks.